npm/extensions.bzl

```diff
@@ -7,58 +7,117 @@ load("//npm/private:npm_translate_lock.bzl", "DEFAULT_REGISTRY", npm_translate_l
 load("//npm/private:npm_import.bzl", npm_import_lib = "npm_import")
 load("//npm:npm_import.bzl", "npm_import", "npm_translate_lock")
 load("//npm/private:transitive_closure.bzl", "translate_to_transitive_closure")
+load("//npm/private:npmrc.bzl", "parse_npmrc")
 
 def _extension_impl(module_ctx):
     for mod in module_ctx.modules:
         for attr in mod.tags.npm_translate_lock:
             lockfile = utils.parse_pnpm_lock(module_ctx.read(attr.pnpm_lock))
             trans = translate_to_transitive_closure(lockfile, attr.prod, attr.dev, attr.no_optional)
 
-            # TODO: this feature introduced in https://github.com/aspect-build/rules_js/pull/503
-            # but not added to bzlmod. For now, not supported here.
             registries = {}
-            imports = npm_translate_lock_lib.gen_npm_imports(trans, attr.pnpm_lock.package, attr, registries, DEFAULT_REGISTRY)
+            default_registry = DEFAULT_REGISTRY
+            npm_tokens = {}
+            if attr.npmrc:
+                npmrc_path = module_ctx.read(attr.npmrc)
```

Comment:
this looks wrong; guessing it should be
```
        npmrc_path = module_ctx.path(module_ctx.attr.npmrc)
        npmrc = parse_npmrc(module_ctx.read(npmrc_path))
```

Reply:
we should also make bzlmod variations of all of the `npm_translate_lock*` e2e tests so we have coverage for both the WORKSPACE and the bzlmod code paths

Reply:
In a module extension the `module_ctx` doesn't have `attrs`; that needs to be grabbed from the tag classes.